library/Garp/Util/AssetUrl.php

```diff
@@ -83,7 +83,7 @@ protected function _getExtension($file) {
     }
 
     public function getVersionedQuery($file) {
-        return $file . '?' . new Garp_Semver();
+        return $file . '?' . new Garp_Version();
     }
 
     public function getVersionedBuildPath($file) {
```

Comment:
Shouldn't we just deprecate this function? Can't imagine this version being used with an old setup like that, if it still exists in the first place 👍

Reply:
Hmmm. Maybe so. It would be a good time to simplify `AssetUrl` a bit.

I actually couldn't be bothered to remove `Garp_Semver` and its tests because it's still a valid implementation of that scheme. 
Would you feel better if it would be gone completely?

Reply:
> Hmmm. Maybe so. It would be a good time to simplify AssetUrl a bit.

That's probably my main point. I might've contributed to that overcomplexity in the past myself, so would be nice to remove legacy stuff that might make the `AssetUrl` any harder to grasp than needed. As far as that class is complex anyway. Could also take a look myself, when I update this pull request with the necessary Capistrano changes of course!

> I actually couldn't be bothered to remove Garp_Semver and its tests because it's still a valid implementation of that scheme. Would you feel better if it would be gone completely?

Sure; doesn't bother me at all. Was probably more a notice that those were still around, and if you're not going to use it, why not remove it?

PS: but awesome work by the way, hopefully my 'changes requested' didn't convey any other message 🙈

Reply:
Haha, no worries, it's a fair point. Why keep it around? I'm always reluctant to trash code that might be useful someday, but there's always a cognitive cost. Someone might find it and think it's the way we do things.

So, maybe I should just throw it out. 
I'll also simplify `AssetUrl` 👍  It's a beast, and this would be a great time to fix that.